src/Selenium2Library/keywords/_screenshot.py

```diff
@@ -17,49 +17,96 @@ def __init__(self):
 
-        `css` can be used to modify how the screenshot is taken. By default
-        the bakground color is changed to avoid possible problems with
-        background leaking when the page layout is somehow broken.
+        Starting from Selenium2Library 1.8 if ``filename`` contains _{index}_
+        characters, it will be automatically replaced with running index.
+        The running index is unique for each different filename. The absolute
+        path of the saved screenshot is always returned and it does not depend
+        does the ``filename`` contain _{index}_. See example 1 and 2 for more
+        details.
+
+        The _{index}_ uses Python
+        [https://docs.python.org/2/library/string.html#string-formatting|
+        String Formatting] and it is possible to format the _{index}_ with
+        [https://docs.python.org/2/library/string.html#grammar-token-width|width].
+        The example 3 uses width to format the _{index}_ to desired length.
+        The width is also the only tested and supported option for _{index}_.
+        Other formatting options may also work but are not tested/supported.
+
+        If there is a need to write literal _{index}_ or if ``filename``
+        contains _{_ or _}_ characters, then the braces must be doubled. See
+        example 4 for more details.
+
+        Example 1:
+        | ${file1} = | Capture Page Screenshot |
+        | File Should Exist | ${OUTPUTDIR}${/}selenium-screenshot-1.png |
+        | Should Be Equal | ${file1} | ${OUTPUTDIR}${/}selenium-screenshot-1.png |
+        | ${file2} = | Capture Page Screenshot |
+        | File Should Exist | ${OUTPUTDIR}${/}selenium-screenshot-2.png |
+        | Should Be Equal | ${file2} | ${OUTPUTDIR}${/}selenium-screenshot-2.png |
+
+        Example 2:
+        | ${file1} = | Capture Page Screenshot | ${OTHER_DIR}${/}other-{index}-name.png |
+        | ${file2} = | Capture Page Screenshot | ${OTHER_DIR}${/}some-other-name-{index}.png |
+        | ${file3} = | Capture Page Screenshot | ${OTHER_DIR}${/}other-{index}-name.png |
+        | File Should Exist | ${OTHER_DIR}${/}other-1-name.png |
+        | Should Be Equal | ${file1} | ${OTHER_DIR}${/}other-1-name.png |
+        | File Should Exist | ${OTHER_DIR}${/}some-other-name-1.png |
+        | Should Be Equal | ${file2} | ${OTHER_DIR}${/}some-other-name-1.png |
+        | File Should Exist | ${OTHER_DIR}${/}other-2-name.png |
+        | Should Be Equal | ${file3} | ${OTHER_DIR}${/}other-2-name.png |
+
+        Example 3:
+        | Capture Page Screenshot | ${OTHER_DIR}${/}sc-{index:06}.png |
+        | File Should Exist | ${OTHER_DIR}${/}sc-000001.png |
+
+        Example 4
+        | Capture Page Screenshot | ${OUTPUTDIR}${/}braces-{{index}}.png |
+        | File Should Exist | ${OUTPUTDIR}${/}braces-{index}.png |
+        | Capture Page Screenshot | ${OUTPUTDIR}${/}brace-{{index.png |
+        | File Should Exist | ${OUTPUTDIR}${/}brace-{index.png |
```

Comment:
I agree that it is more or less copy&paste what the Python docs do say. In the other hand, I do feel that not many will actually read the Python docs that carefully (if at all).

Perhaps good solution is to remove the examples and keep the note in the doc. The kw doc is in any case quite big now.